client/my-sites/plans/main.jsx

```diff
@@ -157,16 +162,65 @@ class Plans extends Component {
 		);
 	}
 
+	renderEcommerceTrialPage() {
+		const { selectedSite, translate, canAccessPlans, domainAndPlanPackage } = this.props;
+		const description = translate(
+			'See and compare the features available on each WordPress.com plan.'
+		);
+
+		return (
+			<div>
+				{ selectedSite.ID && <QuerySitePurchases siteId={ selectedSite.ID } /> }
+				<DocumentHead title={ translate( 'Plans', { textOnly: true } ) } />
+				<PageViewTracker path="/plans/:site" title="Plans" />
+				<QueryContactDetailsCache />
+				<QueryPlans />
+				<TrackComponentView eventName="calypso_plans_view" />
+				<Main wideLayout>
+					{ ! canAccessPlans && (
+						<EmptyContent
+							illustration="/calypso/images/illustrations/illustration-404.svg"
+							title={ translate( 'You are not authorized to view this page' ) }
+						/>
+					) }
+					{ canAccessPlans && (
+						<>
+							<FormattedHeader
+								brandFont
+								headerText={ translate( 'Plans' ) }
+								subHeaderText={ description }
+								align="left"
+							/>
+							{ domainAndPlanPackage && <DomainAndPlanUpsellNotice /> }
+							<div id="plans" className="plans plans__has-sidebar">
+								<PlansNavigation path={ this.props.context.path } />
+								<div>eCommerce Trial special content</div>
+							</div>
+						</>
+					) }
+				</Main>
+			</div>
+		);
+	}
+
 	render() {
 		const { selectedSite, translate, canAccessPlans, currentPlan, domainAndPlanPackage } =
 			this.props;
 
+		const currentPlanSlug = selectedSite.plan.product_slug;
+		const isEcommerceTrial = currentPlanSlug === PLAN_ECOMMERCE_TRIAL_MONTHLY;
+
 		if ( ! selectedSite || this.isInvalidPlanInterval() || ! currentPlan ) {
 			return this.renderPlaceholder();
 		}
 		const description = translate(
 			'See and compare the features available on each WordPress.com plan.'
 		);
+
+		if ( isEcommerceTrial ) {
+			return this.renderEcommerceTrialPage();
```

Comment:
It feels like it would be much cleaner to have `this.renderEcommerceTrialPage()` called via conditional code where we currently call `this.renderPlansMain()` so we're keeping all of the existing logic in place.

Reply:
Also, it may help to have the new function simply call the existing function (i.e. have it call `this.renderPlansMain()`) possibly with a light wrapper div, as we then won't have really ugly placeholder content visible.

Reply:
Thanks, I left it in a way that can be merged without impacting any user.